src/ModelContextProtocol.Analyzers/CS1066Suppressor.cs

```diff
@@ -0,0 +1,148 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace ModelContextProtocol.Analyzers;
+
+/// <summary>
+/// Suppresses CS1066 warnings for MCP server methods that have optional parameters.
+/// </summary>
+/// <remarks>
+/// <para>
+/// CS1066 is issued when a partial method's implementing declaration has default parameter values.
```

Comment:
@stephentoub I think this could be addressed simply by having the source generated copy omit all default parameters.

Reply:
Yup.
cc: @madstorgersen